lib/spack/spack/package.py

```diff
@@ -938,8 +962,11 @@ def build_process():
                 # Run the pre-install hook in the child process after
                 # the directory is created.
                 spack.hooks.pre_install(self)
-
-                if fake:
+                # Set up process's build environment before running install.
```

Comment:
Should probably also emit a different message from `tty.msg("Building %s" % self.name)` on line 954 above.  User should be told that a binary cache is being installed instead.